torch_np/_wrapper.py

```diff
@@ -445,48 +495,52 @@ def size(a, axis=None):
 ###### shape manipulations and indexing
 
 
-def expand_dims(a, axis):
-    a = asarray(a)
+@normalizer
+def expand_dims(a: ArrayLike, axis):
     shape = _util.expand_shape(a.shape, axis)
-    tensor = a.get().view(shape)  # never copies
-    return ndarray._from_tensor_and_base(tensor, a)
+    tensor = a.view(shape)  # never copies
+    return _helpers.array_from(tensor, a)
 
 
-@asarray_replacer()
-def flip(m, axis=None):
-    return _flips.flip(m, axis)
+@normalizer
+def flip(m: ArrayLike, axis=None):
+    result = _flips.flip(m, axis)
+    return _helpers.array_from(result)
 
 
-@asarray_replacer()
-def flipud(m):
-    return _flips.flipud(m)
+@normalizer
+def flipud(m: ArrayLike):
+    result = _flips.flipud(m)
+    return _helpers.array_from(result)
 
 
-@asarray_replacer()
-def fliplr(m):
-    return _flips.fliplr(m)
+@normalizer
+def fliplr(m: ArrayLike):
+    result = _flips.fliplr(m)
+    return _helpers.array_from(result)
 
 
-@asarray_replacer()
-def rot90(m, k=1, axes=(0, 1)):
-    return _flips.rot90(m, k, axes)
+@normalizer
+def rot90(m: ArrayLike, k=1, axes=(0, 1)):
+    result = _flips.rot90(m, k, axes)
+    return _helpers.array_from(result)
 
 
-@asarray_replacer()
-def broadcast_to(array, shape, subok=False):
-    _util.subok_not_ok(subok=subok)
-    return torch.broadcast_to(array, size=shape)
+@normalizer
+def broadcast_to(array: ArrayLike, shape, subok: SubokLike = False):
+    result = torch.broadcast_to(array, size=shape)
+    return _helpers.array_from(result)
 
 
 from torch import broadcast_shapes
 
 
 # YYY: pattern: tuple of arrays as input, tuple of arrays as output; cf nonzero
-def broadcast_arrays(*args, subok=False):
-    _util.subok_not_ok(subok=subok)
-    tensors = _helpers.to_tensors(*args)
-    res = torch.broadcast_tensors(*tensors)
-    return tuple(asarray(_) for _ in res)
+@normalizer
+def broadcast_arrays(*args: UnpackedSeqArrayLike, subok: SubokLike = False):
+    args = args[0]  # undo the *args wrapping in normalizer
```

Comment:
We should hopefully be able to fix this with a proper preprocessing of variadic inputs in the normalizer.

Reply:
Done in https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/70/commits/9d75cab63adaad254996e27f4a927ee4591d0ff0

Reply:
Ah, right, I forgot we need to error out on those. In my head I always thought of the non-error case.

Also, I appreciate it, but no need to attribute the commit really :)